File: IN1010/Oblig/oblig1/Celle.java
class Celle {

    // opretter instansvariabelene for cellens informasjon
     private boolean levende; // 
     private Celle[] naboer; // array som skal holde referenase til alle naboer
     private int antNaboer; // hvor mange naboer cellen har
     private int antLevendeNaboer; // hvor mange av dem lever.



    // kontruktør, oppretter cellen som død og setter array til 8 celleobjekter stor. 
    public Celle(){


        levende = false;
        naboer = new Celle[8];
        antNaboer = 0;
        antLevendeNaboer  = 0;



    }

    // metode for å sette levende status til død
    public void settDoed(){

        levende = false;
    }

    // metode for å sette levende statis til levende
    public void settLevende(){

        levende = true;
    }

    // metode for å sjekke statusen til cellen.
    public boolean erLevende(){

        return levende;

    }


    // For å representere om cellen er levende eller død, returneres en char der: O er levende og . er død
    public char hentStatusTegn(){

        if (levende) {
            return 'O';
            
        } else {
            return '.';
        }
    }

    // Legger til oppgit nabo i naboer arrayen til cellen. 
    public void leggTilNabo(Celle naboen) {

        for (int i = 0; i < naboer.length; i++) {
            if (naboer[i] == null) { // finner ledig plass i arrayen.
                naboer[i] = naboen;
                antNaboer++;
                return;  // avbryter for loopen slik at ikke hele fylles med referanser til samme objekt.
                
            }
        }
    }

    // sjekker hvor mange naboer i arrayen naboer som har status levende. Oppdaterer instansvariabelen antLevendeNaboer.
    public void tellLevendeNaboer() {

        int teller = 0;
        for (int i = 0; i < naboer.length; i++) {
        if (naboer[i] != null){
            if (naboer[i].erLevende()) {
                teller++; 
            }
        }
        antLevendeNaboer = teller;
        }
    }

    // "Reglene for spillet", disse som bestemmer om en celle skal dø, fortsette å leve eller komme tilbake i live. Kjøres en gang per celle for hver generasjon.
    public void oppdaterStatus() {

        if (levende) {
            if (antLevendeNaboer < 2 || antLevendeNaboer > 3 ) {
                settDoed();
            }
        } else {
            if (antLevendeNaboer == 3) {
                settLevende();
            }
        }

}

}

File: IN1000/Obliger/Oblig8/old/kolonne først/celle.py
class Celle:
    # Konstruktør
    def __init__(self):
        self._status = "doed"
        self._naboer = []
        self._ant_levende_naboer = 0

    def sett_doed(self):
        self._status = "doed"

    def sett_levende(self):
        self._status = "levende"

    def legg_til_nabo(self, nabo):
        self._naboer.append(nabo)

    def er_levende(self):
        if self._status == "levende":
            return True
        else:
            return False

    def hent_status(self):
        pass

    def hent_status_tegn(self):
        if self.er_levende():
            return "O"
        else:
            return "."

    def tell_levende_naboer(self):

        teller = 0

        #print(self._naboer) #<----
        for elem in self._naboer:
            if elem.er_levende():
                teller += 1

        #print(teller) #<----
        self._ant_levende_naboer = teller

    def oppdater_status(self):

        if self.er_levende():
    

            if self._ant_levende_naboer < 2:
                self.sett_doed()
            elif self._ant_levende_naboer > 3:
                self.sett_doed()

        else:

            if self._ant_levende_naboer == 3:
                self.sett_levende()
